### PR TITLE
INT-2162: Migration of settings/admin directory

### PR DESCRIPTION
**Pull request by hbjORbj**

### CODE REVIEW

**Comment by hbjORbj** [comment]
Current dependencies on/for this PR:
* main
  * **PR #123** <a href="https://app.graphite.dev/github/pr/intuita-inc/cal.com-demo/123?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/intuita-inc/cal.com-demo/123?utm_source=stack-comment).

File: apps/web/components/auth/layouts/AdminLayoutAppDir.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ...rest
+}: { children: React.ReactNode } & ComponentProps<typeof Shell>) {
+  const pathname = usePathname();
+  const session = useSession();
+  const router = useRouter();
+
+  // Force redirect on component level
+  useEffect(() => {
+    if (session.data && session.data.user.role !== UserPermissionRole.ADMIN) {
+      router.replace("/settings/my-account/profile");
```
Comment (DmytroHryshyn) [inline apps/web/components/auth/layouts/AdminLayoutAppDir.tsx:24]: does this work in app dir?

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/flags/page.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import Page from "@pages/settings/admin/flags";
+import { _generateMetadata } from "app/_utils";
+
+export const generateMetadata = async () =>
+  await _generateMetadata(
+    () => "Feature Flags",
```
Comment (grzpab) [inline apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/flags/page.tsx:6]: these are not translated?

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/page.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Page from "@pages/settings/admin/index";
+import { _generateMetadata } from "app/_utils";
+
+export const generateMetadata = async () =>
+  await _generateMetadata(
+    () => "Admin",
```
Comment (grzpab) [inline apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/page.tsx:6]: these are not translated?

File: apps/web/components/auth/layouts/AdminLayoutAppDir.tsx (hunk 1 above)
Comment (grzpab) [reply apps/web/components/auth/layouts/AdminLayoutAppDir.tsx:24]: I think so

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/users/[id]/edit/page.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  } else {
+    // @ts-expect-error Property 'useSuspenseQuery' does not exist on type
+    const [data] = trpc.viewer.users.get.useSuspenseQuery({ userId: input.data.id });
+    const { user } = data;
+    title = `Editing user: ${user.username}`;
+  }
+
+  return await _generateMetadata(
+    () => title,
+    () => "Here you can edit a current user."
```
Comment (grzpab) [inline apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/users/[id]/edit/page.tsx:25]: no translation?

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/page.tsx (hunk 3 above)
Comment (hbjORbj) [reply apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/page.tsx:6]: nope

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/flags/page.tsx (hunk 2 above)
Comment (hbjORbj) [reply apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/flags/page.tsx:6]: nope according to the legacy page.

File: apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/users/[id]/edit/page.tsx (hunk 4 above)
Comment (hbjORbj) [reply apps/web/app/future/(shared-page-wrapper)/(admin-layout)/settings/admin/users/[id]/edit/page.tsx:24]: nope according to the legacy page.

**Pull request closed without merging**